### PR TITLE
mgr/volumes/nfs: drop `type` param during cluster create

### PR DESCRIPTION
**Pull request by mgfritch**
PR #37600 introduced support for both `cephfs` and `rgw` exports
to be configured using a single nfs-ganesha cluster

Fixes: https://tracker.ceph.com/issues/50369
Signed-off-by: Michael Fritch <mfritch@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by liewegas — APPROVED** [review]

**Comment by batrick** [comment]
Pending discussion with @varshar16

**Comment by sebastian-philipp** [comment]
after a small chat with @guits I somewhat tend to agree with this PR as it makes the deployment a bit easier and flexible. But I might miss something here.

**Comment by varshar16** [comment]
> after a small chat with @guits I somewhat tend to agree with this PR as it makes the deployment a bit easier and flexible. But I might miss something here.

One issue is backward compatibility and other is related to RGW block in [common config](https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/templates/services/nfs/ganesha.conf.j2#L34-L37). https://pad.ceph.com/p/nfs

**Comment by mgfritch** [comment]
> > after a small chat with @guits I somewhat tend to agree with this PR as it makes the deployment a bit easier and flexible. But I might miss something here.
> 
> One issue is backward compatibility 

are there existing consumers of this cli? maybe doc/readme is sufficient?

>and other is related to RGW block in [common config](https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/templates/services/nfs/ganesha.conf.j2#L34-L37). https://pad.ceph.com/p/nfs

.. address the RGW block in another PR?


removing the required positional arg avoids the need to maintain backward compat ... and we could easily re-add it as a required named arg if we discover it's needed later ...

**Comment by mgfritch** [comment]
jenkins test api

**Comment by varshar16** [comment]
> > > after a small chat with @guits I somewhat tend to agree with this PR as it makes the deployment a bit easier and flexible. But I might miss something here.
> > 
> > 
> > One issue is backward compatibility
> 
> are there existing consumers of this cli? maybe doc/readme is sufficient?
> 

I am fine with documenting about it.

> > and other is related to RGW block in [common config](https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/templates/services/nfs/ganesha.conf.j2#L34-L37). https://pad.ceph.com/p/nfs
> 
> .. address the RGW block in another PR?
> 

fine with this too.

File: PendingReleaseNotes [hunk 1]
```diff
@@ -11,6 +11,9 @@
 
 >=16.0.0
 --------
+* volumes/nfs: The ``cephfs`` cluster type has been removed from the
+  ``nfs cluster create`` subcommand.
+
```
Comment (varshar16) [inline PendingReleaseNotes:16]: Add a line saying both ``cephfs`` and ``rgw`` is supported by ``nfs cluster create`` subcommand.

**Review by varshar16 — APPROVED** [review]
Minor nit, otherwise looks good.

File: PendingReleaseNotes (hunk 1 above)
Comment (mgfritch) [reply PendingReleaseNotes]: added verbiage similar to this to the PendingReleaseNotes.

**Comment by batrick** [comment]
Is this to be backported? If so, please create a tracker ticket.

**Comment by batrick** [comment]
jenkins test api

**Comment by mgfritch** [comment]
> Is this to be backported? If so, please create a tracker ticket.

 yeah my bad, tracker is here: https://tracker.ceph.com/issues/50369

I'd be good if we could get this into the next Pacific release.

**Comment by mgfritch** [comment]
jenkins test api

**Comment by varshar16** [comment]
jenkins test api

**Comment by batrick** [comment]
https://tracker.ceph.com/projects/cephfs/wiki/Master#2021-Apr-15

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
ready to merge from cephfs side

**Pull request merged**